unit_tests/src/test_client_connection.cpp

```diff
@@ -300,3 +300,40 @@ TEST_CASE("#7192: all connections store unique session signatures", "client_conn
 
     CHECK(std::all_of(std::begin(pairs), std::end(pairs), has_unique_session_signatures));
 }
+
+TEST_CASE("#7576: ensure no limitaitons on proxied groupadmin", "client_connection")
+{
+    rodsEnv env{};
+    _getRodsEnv(env);
+
+    load_client_api_plugins();
+
+    // Creates a connection to an iRODS server based on the user's
+    // $HOME/.irods/irods_environment.json file.
+    irods::experimental::client_connection admin_conn;
+
+    namespace ia = irods::experimental::administration;
+    const ia::user group_user{"group_user"};
+    REQUIRE_NOTHROW(ia::client::add_user(admin_conn, group_user));
+
+    irods::at_scope_exit remove_defer_user{
+        [&admin_conn, group_user] { ia::client::remove_user(admin_conn, group_user); }};
+
+    REQUIRE_NOTHROW(ia::client::modify_user(admin_conn, group_user, ia::user_type_property{ia::user_type::groupadmin}));
+
+    // Connect to iRODS without authenticating.
+    irods::experimental::client_connection conn{irods::experimental::defer_authentication,
+                                                env.rodsHost,
+                                                env.rodsPort,
+                                                {env.rodsUserName, env.rodsZone}, // (proxy) rodsadmin user
+                                                {"group_user", "tempZone"}}; // groupadmin user
+
+    // Authenticate the rodsadmin.
+    char pw[] = "rods";
```

Comment:
This assumes the test can only be run by a rodsadmin with a password of "rods".

Probably want to create another rodsadmin user and set their password within the test.

Reply:
Done.